uds/diagnostic_configuration/state.py

```
@@ -0,0 +1,85 @@
+        :param name: Value to set (whitespaces will be stripped).
+
+        :raise TypeError: Value must be a string.
+        :raise ValueError: Value must not consist of whitespaces only.
+        """
+        if not isinstance(name, str):
+            raise TypeError(f"Provided name is not str type. Actual type: {type(name)}.")
+        stripped_name = name.strip()
+        if not stripped_name:
+            raise ValueError("Name must not consist of whitespace characters only.")
+        if stripped_name != name:
+            warn(category=UserWarning,
```

Ruff (0.15.20)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)